### PR TITLE
fix(1769): Do not throw err if cannot get changed files

### PR DESCRIPTION
**Pull request by tkyi**
## Context

Sometimes when there are PRs with over 2500 files changed, the webhook will fail since the circuit breaker will timeout.

## Objective

This PR does not throw errs when `getChangedFiles` or `getOpenedPrs` calls fail.

## References

Short term fix for: https://github.com/screwdriver-cd/screwdriver/issues/1769

## License

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -478,7 +478,7 @@ class GithubScm extends Scm {
 
     /**
      * Get a list of names and references of opened PRs
-     * @async  _getOpenedPRs
+     * @async  getOpenedPRs
```
Comment (adong) [inline index.js]: Hmm, `_getOpenedPRs` and `_ getChangedFiles ` are the function names

File: index.js [hunk 2]
```diff
@@ -1236,8 +1238,9 @@ class GithubScm extends Scm {
 
     /**
      * Resolve a pull request object based on the config
-     * @async  getPrRef
+     * @async  getPrInfo
```
Comment (adong) [inline index.js]: `_getPrInfo`

File: index.js [hunk 3]
```diff
@@ -513,7 +513,8 @@ class GithubScm extends Scm {
             }));
         } catch (err) {
             winston.error('Failed to getOpenedPRs: ', err);
-            throw err;
+
+            return [];
```
Comment (adong) [inline index.js]: If we are not returning the error, then we can `return [];` in finally block

File: index.js (hunk 1 above)
Comment (tkyi) [reply index.js]: it's not really consistent in the file 🤷‍♀

File: index.js (hunk 3 above)
Comment (adong) [reply index.js]: I was wrong about the previous comment. Return in `finally` block always override the return value in `try` block.

Maybe we can write something like 

```javascript
        let pullRequests = [];
        try {
           pullRequests = await this.breaker.runCommand({	            const pullRequests = await this.breaker.runCommand({
                action: 'list',	                action: 'list',
                scopeType: 'pulls',	                scopeType: 'pulls',
                token,	                token,
                params: {	                params: {
                    owner,	                    owner,
                    repo,	                    repo,
                    state: 'open'	                    state: 'open'
                }	                }
            });	            });
            return pullRequests.data.map(pullRequest => ({	            return pullRequests.data.map(pullRequest => ({
                name: `PR-${pullRequest.number}`,	                name: `PR-${pullRequest.number}`,
                ref: `pull/${pullRequest.number}/merge`,	                ref: `pull/${pullRequest.number}/merge`,
                username: pullRequest.user.login,	                username: pullRequest.user.login,
                title: pullRequest.title,	                title: pullRequest.title,
                createTime: pullRequest.created_at,	                createTime: pullRequest.created_at,
                url: pullRequest.html_url,	                url: pullRequest.html_url,
                userProfile: pullRequest.user.html_url	                userProfile: pullRequest.user.html_url
            }));	            }));
        } catch (err) {
            winston.error('Failed to getOpenedPRs: ', err);	            winston.error('Failed to getOpenedPRs: ', err);
            // throw err;
        } finally {
            return pullRequests;
	    }
```

Have a default value. 🤔

Comment (tkyi) [reply index.js]: I think it's better practice not to do `return` in `finally`

getting eslint error `no-unsafe-finally` when I try your suggestion
https://eslint.org/docs/rules/no-unsafe-finally

Comment (adong) [reply index.js]: OK. Lets return in catch.

**Pull request merged**